mzLib/Transcriptomics/Classes/NucleicAcid.cs

```diff
@@ -0,0 +1,393 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Chemistry;
+using Easy.Common.Extensions;
+using Easy.Common.Interfaces;
+using MassSpectrometry;
+using MathNet.Numerics;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace Transcriptomics
+{
+    /// <summary>
+    /// A linear polymer of Nucleic acids
+    /// </summary>
+    public abstract class NucleicAcid : INucleicAcid, IEquatable<NucleicAcid>, IHasMass
+    {
+
+        #region Static Properties
+
+        /// <summary>
+        /// The default chemical formula of the five prime (hydroxyl group)
+        /// </summary>
+        public static readonly ChemicalFormula DefaultFivePrimeTerminus = ChemicalFormula.ParseFormula("O-3P-1");
+
+        /// <summary>
+        /// The default chemical formula of the three prime terminus (hydroxyl group)
+        /// </summary>
+        public static readonly ChemicalFormula DefaultThreePrimeTerminus = ChemicalFormula.ParseFormula("OH");
+
+        #endregion
+
+        #region Constuctors
+
+        protected NucleicAcid(string sequence) 
+            : this(sequence, DefaultFivePrimeTerminus, DefaultThreePrimeTerminus)
+        {
+        }
+
+        protected NucleicAcid(string sequence, IHasChemicalFormula fivePrimeTerm, IHasChemicalFormula threePrimeTerm)
+        {
+            MonoisotopicMass = 0;
+            Length = sequence.Length;
+            _nucleicAcids = new Nucleotide[Length];
+            ThreePrimeTerminus = threePrimeTerm;
+            FivePrimeTerminus = fivePrimeTerm;
+            ParseSequence(sequence);
+        }
+
+        #endregion
+
+        #region Private Properties
+
+        /// <summary>
+        /// The 5-Prime chemical formula cap. This is different from the 5-prime terminus modification.
+        /// </summary>
+        private IHasChemicalFormula _5PrimeTerminus;
+
+        /// <summary>
+        /// The 3-Prime chemical formula cap. This is different from the 3-prime terminus modification.
+        /// </summary>
+        private IHasChemicalFormula _3PrimeTerminus;
+
+        /// <summary>
+        /// All of the modifications indexed by position from 5- to 3-prime termini. This array is 2 bigger than the nucleic acid array
+        /// as index 0 and Count - 1 represent the 5- and 3-prime terminus, respectively
```

Comment:
? where count = length + 2?

Reply:
Ignore for now. This was from the CSMSL implementation and is currently unused. It will be changed to be more in line with out standard method when the modification PR comes around

Reply:
we should discuss number (we have the "one is n terminus") business for peptides